main.cpp

```diff
@@ -7,42 +7,371 @@
+			cout << "the move is not correct" << endl;
+			return false;
+		}
+		if (get_col() == dst_col) {
+			//			if (check_turn_general(0, 1, next)) 
+			return 1;
+		}
+		if (get_row() == dst_row) {
+			//			if (check_turn_general(1, 0, next)) 
+			return 1;
+		}
+		cout << "the move is not correct" << endl;
+		return 0;
+	}
+	char to_char() const {
+		return 'E';
+	}
+};
+class Queen : public Figure {
+public:
+	Queen(int st_row, int st_col, char st_color) : Figure(st_row, st_col, st_color) {}
+	bool check_turn(int dst_row, int dst_col, char dst_color) const {
+		if (get_col() == dst_col && get_row() == dst_row) { cout << "You do not move"; return 0; }
+		if (get_color() == dst_color) { cout << "There are your figure" << endl; return 0; }
+		if (!(dst_col <= 8 && get_col() > 0 && dst_row < 8 && get_row() >= 0)) {
+			cout << "the move is not correct" << endl;
+			return 0;
+		}
+		if (get_col() == dst_col) {
+//			if (check_turn_general(0, 1, next)) 
+				return 1;
+		}
+		if (get_row() == dst_row) {
+//			if (check_turn_general(1, 0, next)) 
+			return 1;
+		}
+		if (abs(get_col() - dst_col) == abs(get_row() - dst_row)) {
+//			if (check_turn_general(1, 1, next)) 
+			return 1;
+		}
+		cout << "the move is not correct" << endl;
+		return 0;
+	}
+	char to_char() const {
+		return 'Q';
+	}
+};
+class King : public Figure {
+public:
+	King(int st_row, int st_col, char st_color) : Figure(st_row, st_col, st_color) {}
+	bool check_turn(int dst_row, int dst_col, char dst_color) const {
+		if (get_col() == dst_col && get_row() == dst_row) { cout << "You do not move"; return 0; }
+		if (get_color() == dst_color) { cout << "There are your figure" << endl; return 0; }
+		if (!(dst_col < 8 && get_col() >= 0 && dst_row < 8 && get_row() >= 0)) {
+			cout << "the move is not correct" << endl;
+			return 0;
+		}
+		if (abs(get_col() - dst_col) < 2 && abs(get_row() - dst_row) < 2) {
+			return 1;
+		}
+		cout << "the move is not correct" << endl;
+		return 0;
+	}
+	char to_char() const {
+		return 'K';
 	}
 };
 
+
 class Chess {
-	void make_turn();
+	bool make_turn(char cur_color);
 	void print_field();
-	Figure* field[8][8];
+	Figure* field[8][8] = { nullptr };
 public:
 	void start_game();
 	Chess() {
-		Figure* pb1 = new Pawn();
-		Figure* pb2 = new Horse();
-		field[0][0] = pb1;
-		field[0][1] = pb2;
+		char color = 'b';
+		int l = 1;
+		for (int i = 0; i < 8; i += 7) {
+			field[i][0] = new Rook(i, 0, color);
+			field[i][1] = new Horse(i, 1, color);
+			field[i][2] = new Elephant(i, 2, color);
+			field[i][3] = new Queen(i, 3, color);
+			field[i][4] = new King(i, 4, color);
+			field[i][5] = new Elephant(i, 5, color);
+			field[i][6] = new Horse(i, 6, color);
+			field[i][7] = new Rook(i, 7, color);
+			for (int j = 0; j < 8; j++) {
+				field[i+l][j] = new Pawn(i+l, j, color);
+			}
+			l = -1;
+			color = 'w';
+		}
 	}
 };
 
+bool Chess::make_turn(char cur_color) {
+	while (true) {
+		int col_t, row_t, col_n, row_n;
+		char color_next;
```

Comment:
You might want to add some message for the user here, i.e. "Input your next move, e.g. e2 e4" (and would be good if you actually support user input in that form, since it's easier for chess-players to think in that format